modules/gamAdServerSubmodule.js

```diff
@@ -0,0 +1,27 @@
+import { GAM_VENDOR } from './videoModule/constants/vendorCodes.js';
+import { adServerDirectory } from './videoModule/vendorDirectory.js';
+
+/**
+ * @constructor
+ * @param {Object} dfpModule_ - the DFP ad server module
+ * @returns {AdServerProvider}
+ */
+function GamAdServerProvider(dfpModule_) {
```

Comment:
The intention is for this to be a general ad server interface. It should be agnostic to the video subsystem. WHy do you say it assumes the presence of the video module ? Is it because of the imports from the videoModule folder ?
I'm not sure I understand the suggestion of using a `getAdTagUrl` config for the video module. We would expect the publisher to provide the implementation for getting an ad tag ?

Reply:
what I'm thinking is if the only adserver-specific item you need now is `getAdTagUrl`, and GAM would work for most cases, we could start by making that the default so that we don't need to try to define a general adserver interface - that seems like a big, separate project.

Reply:
What do you mean GAM would work for most cases ? GAM only works if i'm using GAM as an ad server. Most people use GAM but it doesn't feel right to force or give preference to a specific vendor. AdLoox for example or Spring Serve might want to implement a submodule.

Reply:
Yes, I agree. What I'm saying is, rather than stubbing an ad server interface and providing only a GAM implementation as you do here, I would prefer to make it a video configuration option, with a default that works for GAM. Either way if you're not using GAM it will not work out of the box; but for the end user that is on another adserver, the config is easier to figure out - and we don't run the risk of making the wrong choices on the interface until we think about that explicitly and more generally. 

For example, I'm pretty sure a general ad server interface should be registered under `videModule/vendorDirectory`, because we'll definitely have situations where we need to understand the ad server but we don't have the video module installed. But I don't know yet where or how it should be registered, I think we can kick that can down the road and tackle it separately.

Reply:
@dgirardi I think I see your point. I agree that we can move the ad server components out of the videoModule folder; i'll address asap. 
As far as providing a video config option instead of an ad server module, I'm leaning towards keeping the ad server modules and modifying the ad server interface in the future. The goal is to make integration as easy as possible for the publisher; I think it's easier for them to name their ad server instead of coming up with the right implementation. 
Since the Video Module is the only client of the adServer interface, I'm fine with it changing as we get more input from ad servers and other potential clients.
To be clear, when configuring prebid and the video module, all the publisher has to provide is:
```
adServer: {
      vendorCode: "gam"
}
```
and optionally they can provide a `params` block to include any additional params that they wish to be used when the ad tag is generated. 
This is a lot simpler than the pub providing an implementation in a config.